src/resource/ResourceLoader.cpp

```diff
@@ -40,21 +42,53 @@ std::shared_ptr<Window> ResourceLoader::GetContext() {
 }
 
 std::shared_ptr<Resource> ResourceLoader::LoadResource(std::shared_ptr<OtrFile> fileToLoad) {
+    auto stream = std::make_shared<MemoryStream>(fileToLoad->Buffer.get(), fileToLoad->BufferSize);
+    auto reader = std::make_shared<BinaryReader>(stream);
     std::shared_ptr<Resource> result = nullptr;
+    ResourceType resourceType;
+    uint64_t id = 0xDEADBEEFDEADBEEF;
 
-    if (fileToLoad != nullptr) {
-        auto stream = std::make_shared<MemoryStream>(fileToLoad->Buffer.data(), fileToLoad->Buffer.size());
-        auto reader = std::make_shared<BinaryReader>(stream);
+    // Determine if file is binary or XML...
+    uint8_t firstByte = reader->ReadInt8();
 
+    if (firstByte == '<') 
+    {
+        // XML
+        reader->Seek(-1, SeekOffsetType::Current);
```

Comment:
If it's xml, we should probably still include the OTR header 64 bytes, even if it's unused. Everything that goes in the archive should have an OTR header.

I'm thinking the best design here is to add a new resource type in LUS called XML, and then reading the cstring as part of that code.